File: TestProj/Assets/juiceScript.js
//add some "juice" and screenshake to the game
var cameraDrift : float = 0.016;
var screenshake : float = 0.005;
var returnSpeed : float = 0.015;
//Balancing stuff, TOUCH CAREFULLY
var zoomSpeed : float = 0.01;
var spinSpeed : float = 0.1;
//misc vars
var hit : boolean = false;
var shakeTime : float = 0.2;
private var timer : float = 0;
private var player : GameObject;
function Start(){
	player = GameObject.FindGameObjectWithTag("Player");
}
function Update(){
	hitCheck();
	this.transform.position = Vector3.MoveTowards(this.transform.position,Vector3.zero,returnSpeed);
	if(hit == true){
		timer+=Time.deltaTime;
		this.transform.position = this.transform.position +Random.insideUnitSphere * screenshake;
		if(timer >= shakeTime){
			hit = false;
			timer = 0;
		}
	}
	this.transform.position = Vector3.MoveTowards(this.transform.position,player.transform.position,cameraDrift);
	Zoom();
	this.transform.position.z = -10;
	Quit();
}
function Zoom(){
	if(this.GetComponent("Camera").orthographicSize > 0.8){
		this.GetComponent("Camera").orthographicSize -=zoomSpeed;
	}
}
function hitCheck(){
	hit = player.GetComponent("player").hit;
}
function Quit(){
	if(Input.GetKey("escape")){
		Application.Quit();
	}
}

File: TestProj/Assets/juiceScript.cs
using UnityEngine;
using System.Collections;

[System.Serializable]
public partial class juiceScript : MonoBehaviour
{
    //add some "juice" and screenshake to the game
    public float cameraDrift;
    public float screenshake;
    public float returnSpeed;
    //Balancing stuff, TOUCH CAREFULLY
    public float zoomSpeed;
    public float spinSpeed;
    //misc vars
    public bool hit;
    public float shakeTime;
    private float timer;
    private GameObject player;
    public virtual void Start()
    {
        player = GameObject.FindGameObjectWithTag("Player");
    }

    public virtual void Update()
    {
        hitCheck();
        transform.position = Vector3.MoveTowards(this.transform.position, Vector3.zero, this.returnSpeed);
        if (this.hit == true)
        {
            this.timer = this.timer + Time.deltaTime;
            this.transform.position = this.transform.position + (Random.insideUnitSphere * this.screenshake);
            if (this.timer >= this.shakeTime)
            {
                this.hit = false;
                this.timer = 0;
            }
        }
        this.transform.position = Vector3.MoveTowards(this.transform.position, this.player.transform.position, this.cameraDrift);
        this.Zoom();

        {
            int _7 = -10;
            Vector3 _8 = this.transform.position;
            _8.z = _7;
            this.transform.position = _8;
        }
        this.Quit();
    }

    public virtual void Zoom()
    {
        if (GetComponent<Camera>().orthographicSize > 0.8f)
        {
            GetComponent<Camera>().orthographicSize = GetComponent<Camera>().orthographicSize - zoomSpeed;
        }
    }

    public virtual void hitCheck()
    {
        hit = player.GetComponent<Player>().hit;
    }

    public virtual void Quit()
    {
        if (Input.GetKey("escape"))
        {
            Application.Quit();
        }
    }

    public juiceScript()
    {
        cameraDrift = 0.016f;
        screenshake = 0.005f;
        returnSpeed = 0.015f;
        zoomSpeed = 0.01f;
        spinSpeed = 0.1f;
        shakeTime = 0.2f;
    }

}